tvm/python/tvm/relay/op/strategy/adreno.py

```diff
@@ -43,12 +42,13 @@ def conv2d_strategy_adreno(attrs, inputs, out_type, target):
                 name="conv2d_nchw.cuda",
             )
         elif data_layout == "NCHW4c" and kernel_layout == "OIHW4o":
-            strategy.add_implementation(
-                wrap_compute_conv2d(topi.adreno.conv2d_nchwc),
-                wrap_topi_schedule(topi.adreno.schedule_conv2d_nchwc),
-                name="conv2d_nchwc.image2d",
-                plevel=10
-            )
+            if out_type.dtype == "float16":
+                strategy.add_implementation(
+                    wrap_compute_conv2d(topi.adreno.conv2d_nchwc),
+                    wrap_topi_schedule(topi.adreno.schedule_conv2d_nchwc),
+                    name="conv2d_nchwc.image2d",
+                    plevel=10
+                )
             strategy.add_implementation(
```

Comment:
Is my understanding correct that we don't have else block for dtype = "float16" because we will be adding acc32 variant of topi schedule both when dtype = "float16" for fp16_acc32 mode and dtype != "float16" = "float32 for plain vanilla fp32 mode?

Reply:
right. I am planning to change the logic of acc16/acc32 soon
I am planning to ad explicit transformation of the graph which will add attribute of the accumulator type. In our branches we will have to call it manually in evaluate.py, but in master it will be called withing fp16 mixed precision transformation